common/plugins/eu.esdihumboldt.hale.common.filter.test/src/eu/esdihumboldt/hale/common/filter/FilterTest.java

```diff
@@ -129,61 +133,82 @@ public void simpleFilterTestCQL() throws CQLException {
 
 	@Test
 	public void simpleSchemaTestCQL() throws Exception {
-		ShapeSchemaReader schemaReader = new ShapeSchemaReader();
-		schemaReader.setSource(new DefaultInputSupplier(getClass().getResource(
-				"/testdata/GN_Point/GN_Point.shp").toURI()));
+		// Specify the path of the resource
+		String resourcePath = "/testdata/GN_Point/GN_Point.zip";
+		Path tempDirectory = java.nio.file.Files.createTempDirectory("extracted-zip");
 
-		schemaReader.validate();
-		IOReport report = schemaReader.execute(null);
-		assertTrue(report.isSuccess());
-
-		Schema schema = schemaReader.getSchema();
+		try {
+			Path shpTempFile = IOUtils.withTemporaryExtractedZipResource(resourcePath,
+					FilterTest.class, FilterTest::processContents, tempDirectory);
```

Comment:
You don't need to use a method for the functional parameter, you can define it inline.

I would suggest to add the whole logic that requires the temporary files to be present to add to the function and leave the creation and the cleanup of the temporary folder to the `withTemporaryExtractedZipResource` method. Otherwise you could have just used `IOUtils.extract` directly.

For example:

```
IOUtils.withTemporaryExtractedZipResource(resourcePath, FilterTest.class, (Path tempFolder) -> {
  //TODO process extracted Shapefile
});
```

Reply:
@stempler 
I'm facing a situation where I cannot include the temporary files in the `IOUtils.withTemporaryExtractedZipResource` because the folder needs to be cleaned after the extracted files are used. The lines below illustrate how the source is set using the temporary file:
`schemaReader.setSource(new DefaultInputSupplier(shpTempFile.toUri()));`
`instanceReader.setSource(new DefaultInputSupplier(shpTempFile.toUri()));`

However, using only `IOUtils.extract` is not an option either, as it would require spreading the logic across three different places. 

What would you recommend or prefer in this scenario?

Reply:
The idea with using the function as parameter as suggested was to do all the processing that requires the files to be present in the passed-in function/lambda, so that `withTemporaryExtractedZipResource` can handle all the hassle with creating and deleting the temporary folder transparently.

Reply:
I understand. Are you suggesting that the logic should be relocated back to each individual test?